src/utils/utils.js

```diff
@@ -0,0 +1,29 @@
+import { getRequest } from "../network/axiosClient";
+import { toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
+export const fetchUsers = async (data, url) => {
+  const users = await getRequest(url);
+  data(users.data);
+};
+
+export const searchUsers = (query, users) => {
+  return users.filter((user) =>
+    user.name.toLowerCase().includes(query.toLowerCase())
+  );
+};
+
+export const filterPosts = (posts, id) => {
+  return posts.filter((item) => item.userId === parseInt(id));
+};
+
+export const errorHandler = error => {
+  if (error.response.status === 404) {
```

Comment:
500 are internal server error when server is crashes or can't handle the request gracefully and in those cases we show "Something went wrong, please try again later!"

Reply:
Usually 404 is returned when content doesn't exist, and we communicate that to user as well.